compiler/rustc_mir_build/src/build/mod.rs

```diff
@@ -725,6 +733,7 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             var_indices: Default::default(),
             unit_temp: None,
             var_debug_info: vec![],
+            lint_level_roots_cache: FxHashSet::default(),
```

Comment:
Should we initialize with `self.hir_id`?

Reply:
No. Due to the `parent_id == self.hir_id` test on the second call, `self.hir_id` is never passed to `maybe_lint_level_root_bounded` :)